src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -275,22 +322,61 @@ def _getNumberOfTiles(self):
                      for s, f in zip(self._area_size, reliable_fov)]
         nx = math.ceil(area_size[0] / reliable_fov[0])
         ny = math.ceil(area_size[1] / reliable_fov[1])
-        logging.debug("Calculated number of tiles nx= %s, ny= %s" % (nx, ny))
 
         # We have a little bit more tiles than needed, we then have two choices
         # on how to spread them:
         # 1. Increase the total area acquired (and keep the overlap)
         # 2. Increase the overlap (and keep the total area)
         # We pick alternative 1 (no real reason)
-        center = (self._area[0] + self._area[2]) / 2, (self._area[1] + self._area[3]) / 2
-        total_size = nx * reliable_fov[0], ny * reliable_fov[1]
-
-        # Compute the top-left of the "bigger" area, and from it, shift by half
-        # the size of the smallest tile.
-        starting_pos = {'x': center[0] - total_size[0] / 2 + reliable_fov[0] / 2,  # left
-                        'y': center[1] + total_size[1] / 2 - reliable_fov[1] / 2}  # top
-
-        return (nx, ny), starting_pos
+        xmin, ymin, xmax, ymax = self._polygon.bounds
+        center = ((xmin + xmax) / 2, (ymin + ymax) / 2)
+        total_size = (
+            nx * reliable_fov[0] + self._sfov[0] * self._overlap,
+            ny * reliable_fov[1] + self._sfov[1] * self._overlap,
+        )
+        xmin = center[0] - total_size[0] / 2
+        ymax = center[1] + total_size[1] / 2
+
+        # Create an empty grid for storing intersected tiles
+        # An intersected tile is any tile in the grid that intersects with (or falls within) the given polygon
+        tile_grid = numpy.zeros((ny, nx), dtype=bool)
+
+        # Vectorized conversion of polygon points to grid coordinates
+        points = numpy.array(self._polygon.exterior.coords)
+        rows = numpy.floor((ymax - points[:, 1]) / reliable_fov[1]).astype(int)
+        cols = numpy.floor((points[:, 0] - xmin) / reliable_fov[0]).astype(int)
+
+        # Vectorized calculation of intersected grid cells
+        row_pairs = numpy.vstack((rows, numpy.roll(rows, -1))).T
+        col_pairs = numpy.vstack((cols, numpy.roll(cols, -1))).T
+
+        intersected_tiles = get_possible_intersections(row_pairs, col_pairs)
```

Comment:
Can add the `include_neighbors` argument explicitly, it'll be a little clearer on what it does, and what to expect as result.